csrc/instance_norm_nvfuser_kernel.cu

```diff
@@ -0,0 +1,279 @@
+      const auto _input_dtype = get_dtype(input.scalar_type());
+      const auto _grad_output_dtype = get_dtype(grad_output.scalar_type());
+      const auto _weight_dtype = get_dtype(weight.scalar_type());
+      const auto _running_mean_dtype = get_dtype(run_mean.scalar_type());
+      const auto _running_var_dtype = get_dtype(run_var.scalar_type());
+      auto _input = makeContigTensor(input.sizes().size(), _input_dtype);
+      auto _grad_output = makeContigTensor(grad_output.sizes().size(), _grad_output_dtype);
+      auto _weight = makeContigTensor(weight.sizes().size(), _weight_dtype);
+      auto _running_mean = makeContigTensor(run_mean.sizes().size(), get_dtype(run_mean.scalar_type()));
+      auto _running_var = makeContigTensor(run_var.sizes().size(), get_dtype(run_var.scalar_type()));
+      auto _save_mean = makeContigTensor(save_mean.sizes().size(), get_dtype(save_mean.scalar_type()));
+      auto _save_invstd = makeContigTensor(save_invstd.sizes().size(), get_dtype(save_invstd.scalar_type()));
+
+      fusion->addInput(_input);
+      fusion->addInput(_grad_output);
+      fusion->addInput(_weight);
+      fusion->addInput(_running_mean);
+      fusion->addInput(_running_var);
+      fusion->addInput(_save_mean);
+      fusion->addInput(_save_invstd);
+
+      if (_input_dtype == DataType::Half || _input_dtype == DataType::BFloat16) {
+        _input = castOp(DataType::Float, _input);
+      }
+      if (_grad_output_dtype == DataType::Half || _grad_output_dtype == DataType::BFloat16) {
+        _grad_output = castOp(DataType::Float, _grad_output);
+      }
+      if (_weight_dtype == DataType::Half || _weight_dtype == DataType::BFloat16) {
+        _weight = castOp(DataType::Float, _weight);
+      }
+      if (_running_mean_dtype == DataType::Half || _running_mean_dtype == DataType::BFloat16) {
+        _running_mean = castOp(DataType::Float, _running_mean);
+      }
+      if (_running_var_dtype == DataType::Half || _running_var_dtype == DataType::BFloat16) {
+        _running_var = castOp(DataType::Float, _running_var);
+      }
+
+      Double* _eps = IrBuilder::create<Double>();
+      fusion->addInput(_eps);
+      if (!run_mean.sizes().size()) {
+        _running_mean = nullptr;
+        _running_var = nullptr;
+      } 
+      if (!weight.sizes().size()) {
+        _weight = nullptr;
+      }
+      auto result = instance_norm_backward(_input,
+                                           _grad_output,
+                                           _weight,
+	  				                       _running_mean,
+                                           _running_var,
+	  				                       _save_mean,
+                                           _save_invstd,
+                                           use_input_stats,
+                                           _eps,
+                                           {true, true, true}, // TODO: is output mask useful?
+                                           channels_last);
```

Comment:
[nit-picking] could you lint these lines as it seems like you've interchangeably used spaces and tabs...